sklearn/model_selection/_split.py

```diff
@@ -2154,6 +2155,205 @@ def get_n_splits(self, X=None, y=None, groups=None):
         return len(self.unique_folds)
 
 
+class GroupTimeSeriesSplit(_BaseKFold):
+    """Time Series cross-validator variant with non-overlapping groups.
+
+    Provides train/test indices to split time series data samples
+    that are observed at fixed time intervals according to a
+    third-party provided group.
+    In each split, test indices must be higher than before, and thus shuffling
+    in cross validator is inappropriate.
+
+    This cross-validation object is a variation of :class:`KFold`.
+    In the kth split, it returns first k folds as train set and the
+    (k+1)th fold as test set.
+
+    The same group will not appear in two different folds (the number of
+    distinct groups has to be at least equal to the number of folds).
+
+    Note that unlike standard cross-validation methods, successive
+    training sets are supersets of those that come before them.
+
+    The groups should be continuous. For Example:
```

Comment:
```suggestion
    The groups should be contiguous. For Example:
```

Reply:
> I think this meaning of group, as a collection of contiguous samples that make up an individual data point, is a bit different from the meaning in `GroupKFold`. With this definition, you could effectively just treat the groups as samples and run `TimeSeriesSplit` over them, removing the need for a lot of the code here (both implementation and test). I think @scikit-learn/core-devs need to decide if this kind of "contiguous group in time" case is sufficiently general to be included in the scikit-learn library.

@jnothman  - Is this a critique for #16236   ?

Reply:
Yes, the assumption of contiguous groups is a bit special relative to our handling of groups in splitters. OTOH, TimeSeriesSplit is really the only one so far that assumes a lot about the order of samples (except insofar as kfold without shuffling attempts to respect input ordering)

Reply:
Ok . I'll try  make it work for  non-contiguous groups then . I had originally started  on that path and had created a dictionary with group names as keys and corresponding indices as values.